cyblog.md

```diff
@@ -0,0 +1,75 @@
+
+```ts
+// certCertificate and zone created as above
+
+const certValidation = new aws.route53.Record("cert_validation", {
+    records: [certCertificate.domainValidationOptions[0].resourceRecordValue],
+    ttl: 60,
+    type: certCertificate.domainValidationOptions[0].resourceRecordType,
+    zoneId: zone.id,
+});
+```
+
+That's a lot nicer than before!  The following improvements happened:
+1. there's no more lambda!
+2. there's no need for a repetive lambda parameter (which might conflict with some other name in scope).
+3. the code *is* just pure simple *idiomatic* JavaScript/TypeScript that clearly conveys its intent.
+
+Importantly, no information has been lost here.  The exact same dependency information flows along here like it did before.  And, thanks to TypeScript's flexible type system, the above is totally typesafe and will let you know the right types of things and will still error if you happen to make mistakes.
+
+So, how was this done?  Well, the core part of the change is thanks to a little-known feature of JavaScript: (Proxies)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy].
+
+Proxies allow you to return objects where you can control many facets of how JavaScript works at runtime.  There are many things a Proxy lets you control, however for our needs the most important bit was that it allows you to override how member-lookup works.  In other words, when you write `someProxy.someMember` you get a chance to determine what should happen an what `someMember` should actually return!  With that flexible interception point available, we actually took our core `Output` type and made it into a `Proxy`.  We then added the right interception code so that if you ever write `someOutput.someMember` that that gets translated exactly into `someOutput.apply(o => o.someMember)`.   his also works just fine for array-accesses (which are just property-lookups from JavaScript's perspective).  In other words, at runtime, the member-lookup form and the `.apply` form will be equivalent (except that the former is so much nicer to write!).  This is why, for example, `certCertificate.domainValidationOptions[0].resourceRecordValue` will have the correct value (with all the right dependency information).  At runtime it really will be equivalent to the original `certCertificate.apply(certCertificate => certCertificate.domainValidationOptions[0].resourceRecordValue)` form.
+
+Now, while this was fairly easy to get working at runtime from a JavaScript perspective, it was a little more challenging to figure out how to make this work in TypeScript's typing system.  For example, if you had a value like so:
```

Comment:
We started off this post in the third person ("It allows one to pass a piece of code that will be applied to the underlying value...") and ended up addressing the second person here ("For example, if you had a value like so..."). What about using the first person plural for this segment, since we're introducing a hypothetical: "For example, suppose we had a value like so"?